firmware/boron-ins-fsm/src/ins3331.cpp

```diff
@@ -1,180 +1,160 @@
-	// Create the thread
-	new Thread("readINSThread", threadINSReader);  
+os_queue_t insQueue;
 
+// setup function & subfunctions
+void setupINS3331() {
+    // Create a queue. Each element is an unsigned char, there are 25 elements. Last parameter is always 0.
+    os_queue_create(&insQueue, sizeof(rawINSData), 128, 0);
+    // Create the thread
+    new Thread("readINSThread", threadINSReader);
 }
 
-//in the future, checkINS3331() will become a thread
-filteredINSData checkINS3331(){
-
-  rawINSData dataToParse;
-  static CircularBuffer<int,MOVING_AVERAGE_BUFFER_SIZE> iBuffer, qBuffer;
-  static float iSum = 0;
-  static float qSum = 0;
-  static filteredINSData returnINSData = {0,0,0};
-
-  //os_queue_take returns 0 on success, so if we get a value, dump it to circular buffer
-  if (os_queue_take(insQueue, &dataToParse, 0, 0) == 0) {
-
-    //add absolute values to circular buffer for the rolling average
-    //since push() adds to the tail, adding beyond capacity causes the element at head to be overwritten and lost
-    iBuffer.push(abs(dataToParse.inPhase));
-    qBuffer.push(abs(dataToParse.quadrature));
-
-    //compute average of the first n data points by computing the full sum
-    if(iBuffer.size() == MOVING_AVERAGE_SAMPLE_SIZE){
-
-      for (int i = 0; i < MOVING_AVERAGE_SAMPLE_SIZE; i++) {
-        iSum += iBuffer[i];
-        qSum += qBuffer[i];
-      }
-      returnINSData.iAverage = iSum/MOVING_AVERAGE_SAMPLE_SIZE;
-      returnINSData.qAverage = qSum/MOVING_AVERAGE_SAMPLE_SIZE;
-  
-    }
-
-    //compute subsequent averages by using sum = sum - oldVal + newVal
-    //note buffer size must be at least one greater than sample size to retain oldVal
-    if(iBuffer.size() == MOVING_AVERAGE_BUFFER_SIZE){
-
-      /*for (int i = 0; i < MOVING_AVERAGE_BUFFER_SIZE; i++) {
-        Log.info("iBuf[%d], qBuf[%d]: %d,   %d", i,i,iBuffer[i],qBuffer[i]);
-      }
-      */
-      iSum = iSum - iBuffer[0] + iBuffer[MOVING_AVERAGE_BUFFER_SIZE-1];
-      qSum = qSum - qBuffer[0] + qBuffer[MOVING_AVERAGE_BUFFER_SIZE-1];
-      returnINSData.iAverage = iSum/MOVING_AVERAGE_SAMPLE_SIZE;
-      returnINSData.qAverage = qSum/MOVING_AVERAGE_SAMPLE_SIZE;
-      //record the time this value was removed from the queue and calculated
-      returnINSData.timestamp = millis();
-      //Log.info("iAverage = %f, qAverage = %f", returnINSData.iAverage, returnINSData.qAverage); 
-    } 
-
-  } //end queue if
-
-  return returnINSData;
-
-} //end checkINS3331()
-
+// in the future, checkINS3331() will become a thread
+filteredINSData checkINS3331() {
+    rawINSData dataToParse;
+    static CircularBuffer<int, MOVING_AVERAGE_BUFFER_SIZE> iBuffer, qBuffer;
+    static filteredINSData returnINSData = {0, 0, 0};
+
+    // os_queue_take returns 0 on success, so if we get a value, dump it to circular buffer
+    if (os_queue_take(insQueue, &dataToParse, 0, 0) == 0) {
+        static float iSum = 0;
+        static float qSum = 0;
```

Comment:
Why did these move?

Reply:
This was for the same reason as the previous variable, to fix a complaint from cppcheck saying that the scope of iSum and qSum could be reduced